cypress-io/cypress
# chore(webkit): update error stack parsing and related system tests

- Related to #6422

### User facing changelog


N/A

### Additional details


WebKit error stacks currently have a number of documented issues, resulting in stack traces that aren't very helpful:

[https://bugs.webkit.org/show_bug.cgi?id=86493](https://bugs.webkit.org/show_bug.cgi?id=86493)
[https://bugs.webkit.org/show_bug.cgi?id=243668](https://bugs.webkit.org/show_bug.cgi?id=243668)
[https://bugs.webkit.org/show_bug.cgi?id=174380](https://bugs.webkit.org/show_bug.cgi?id=174380)

<img width="664" alt="Screen Shot 2022-09-08 at 11 12 28 AM" src="https://user-images.githubusercontent.com/1711637/189172550-cfe5a905-67ac-4c76-8925-f02e5e8b8ee4.png">

This PR updates WebKit stack lines with some placeholder function/location values, which makes them more readable (if not more functional) and allows us to better parse them for presentation within the reporter and snapshot normalization. We are using `<unknown>` in the existing parsing logic when function names/locations could not be determined and I continued that here, but we could use something more verbose if it would be helpful.

It's important to note that the lack of call locations in these stacks makes a lot of our existing strategies for stack inspection/replacement much harder (if not impossible) to implement. A [number of changes](https://bugs.webkit.org/show_bug.cgi?id=243668#c3) have recently landed in WebKit that should improve these stacks, at least so far as removing a lot of these internal stack items that this PR is working with. But the full impact of these changes (and the timeline for their inclusion in a new playwright-webkit release) is yet to be determined.

### Steps to test


* Open project of choice in WebKit
* Update spec to fail (throw error in hook, `expect(true).to.eq(false)`, throw uncaught error in fixture, etc.)
* View stack trace of error, you shouldn't see any lonesome "@" symbols or stack entries missing a function name or location entirely

### How has the user experience changed?


Before:

![Screen Shot 2022-09-07 at 11 58 14 AM](https://user-images.githubusercontent.com/1711637/189169998-befe33cb-b7cf-49f0-9bed-b2b20578c39a.png)

After:

![Screen Shot 2022-09-08 at 10 26 11 AM](https://user-images.githubusercontent.com/1711637/189171881-a8a76921-ebbc-4945-9487-7ed7ab3a26bd.png)

### PR Tasks


- [n/a] Has the original issue (or this PR, if no issue exists) been tagged with a release in ZenHub? (user-facing changes only)
- [n/a] Has a PR for user-facing changes been opened in [`cypress-documentation`](https://github.com/cypress-io/cypress-documentation)? 
- [n/a] Have API changes been updated in the [`type definitions`](https://github.com/cypress-io/cypress/blob/develop/cli/types/cypress.d.ts)?
